StreamChatClient_v3/Database/DTOs/ChannelDTO.swift

```diff
@@ -95,12 +95,49 @@ extension NSManagedObjectContext {
 extension ChannelDTO {
     static func channelListFetchRequest(query: ChannelListQuery) -> NSFetchRequest<ChannelDTO> {
         let request = NSFetchRequest<ChannelDTO>(entityName: "ChannelDTO")
-        request.sortDescriptors = [.init(key: "cid", ascending: true)] // TODO: sorting from query
+        request.sortDescriptors = query.sort.sortDescriptors
         request.predicate = NSPredicate(format: "ANY queries.filterHash == %@", query.filter.filterHash)
         return request
     }
 }
 
+public extension Array where Element == Sorting {
+    var sortDescriptors: [NSSortDescriptor] {
+        if isEmpty || (count == 1 && first?.field == "updated_at") {
+            return .defaultSortDescriptors(isAscending: first?.isAscending ?? false)
+        }
+        
+        var sortDescriptors = [NSSortDescriptor]()
+        
+        forEach { sorting in
+            var key: String?
+            
+            switch sorting.field {
+            case "cid", "id": key = "cid"
+            case "type", "typeRawValue": key = "typeRawValue"
+            case "created_at", "createdDate": key = "createdDate"
+            case "deleted_at", "deletedDate": key = "deletedDate"
+            case "updated_at", "updatedDate": key = "defaultSortingDate"
+            case "last_message_at", "lastMessageDate": key = "lastMessageDate"
```

Comment:
Why don't we create something like:
```swift
enum ChannelSortingKey {
    case cid
    ...
    case custom(String)
}
```

Reply:
... including
```
enum ChannelSortingKey {
    ...
    case default
    ...
}
```

Reply:
I thought about that when we discussed it at the beginning and I tried, but it doesn't make sense because we already by default has a string sorting value. It's just an internal mapping. It makes sense to do for `Sorting` type and it was like a `KeyPath`, but we have to have a custom value. It was discussed and we simplified this part.
For me the latest implementation is KISS and it's clear how it works.